tests/bdd/features/rebuild/test_log_based_rebuild.py

```diff
@@ -0,0 +1,318 @@
+VOLUME_SIZE = 10485761
+NUM_VOLUME_REPLICAS = 2
+NODE_1_NAME = "io-engine-1"
+NODE_2_NAME = "io-engine-2"
+NODE_3_NAME = "io-engine-3"
+POOL_1_UUID = "4cc6ee64-7232-497d-a26f-38284a444980"
+POOL_2_UUID = "91a60318-bcfe-4e36-92cb-ddc7abf212ea"
+POOL_3_UUID = "4d471e62-ca17-44d1-a6d3-8820f6156c1a"
+RECONCILE_PERIOD_SECS = 1
+FIO_RUNTIME = 120
+faulted_child_uri = None
+
+
+@pytest.fixture(autouse=True)
+def init():
+    Deployer.start(
+        io_engines="3",
+        wait="10s",
+        reconcile_period=f"{RECONCILE_PERIOD_SECS}s",
+        faulted_child_wait_period="60s",
+        cache_period="1s",
+        max_rebuilds=f"{MAX_REBUILDS}",
+    )
+
+    # Only create 2 pools so we can control where the intial replicas are placed.
+    ApiClient.pools_api().put_node_pool(
+        NODE_1_NAME, POOL_1_UUID, CreatePoolBody(["malloc:///disk?size_mb=50"])
+    )
+    ApiClient.pools_api().put_node_pool(
+        NODE_2_NAME, POOL_2_UUID, CreatePoolBody(["malloc:///disk?size_mb=50"])
+    )
+
+    yield
+    Deployer.stop()
+
+
+@scenario("log_based_rebuild.feature", "Fast rebuild of crashed replica pod")
+def test_fast_rebuild_of_crashed_replica_pod():
+    """Fast rebuild of crashed replica pod."""
+
+
+@scenario("log_based_rebuild.feature", "Fast rebuild of offlined replica")
+def test_fast_rebuild_of_offlined_replica():
+    """Fast rebuild of offlined replica."""
+
+
+@scenario(
+    "log_based_rebuild.feature", "Fast rebuild of replica after connection interruption"
+)
+def test_fast_rebuild_of_replica_after_connection_interruption():
+    """Fast rebuild of replica after connection interruption."""
+
+
+@scenario(
+    "log_based_rebuild.feature",
+    "Faulted child is not online again within timed-wait period",
+)
+def test_faulted_child_is_not_online_again_within_timedwait_period():
+    """Faulted child is not online again within timed-wait period."""
+
+
+@scenario(
+    "log_based_rebuild.feature",
+    "Faulted child is online again within timed-wait period",
+)
+def test_faulted_child_is_online_again_within_timedwait_period():
+    """Faulted child is online again within timed-wait period."""
+
+
+@scenario(
+    "log_based_rebuild.feature",
+    "Node goes permanently down while log-based rebuild running",
+)
+def test_node_goes_permanently_down_while_logbased_rebuild_running():
+    """Node goes permanently down while log-based rebuild running."""
+
+
+@scenario(
+    "log_based_rebuild.feature",
+    "Volume target is moved before starting log-based rebuild",
+)
+def test_volume_target_is_moved_before_starting_logbased_rebuild():
+    """Volume target is moved before starting log-based rebuild."""
+
+
+@scenario(
+    "log_based_rebuild.feature", "Volume target is moved during log-based rebuild"
+)
+def test_volume_target_is_moved_during_logbased_rebuild():
+    """Volume target is moved during log-based rebuild."""
+
+
+@given('a replica is taken offline by "OfflineChild" API')
+def _():
```

Comment:
Thease came as-is during file generation using pytest-bdd. Will need to appropriately give a valid name I guess as we implement.

Reply:
that's very odd, never happened to me, did you use `pytest-bdd generate file.feature`?

Reply:
yes I used this command. One thing we'll need to cross-check if this could be due to differences in `pytest-bdd` vs `pytest-3-bdd`. because I think sometimes I used `pytest-3-bdd` too.